Sources/System/FilePath/FilePathString.swift

```diff
@@ -0,0 +1,251 @@
+  ///
+  /// - Parameter body: A closure with a pointer parameter
+  ///   that points to a null-terminated platform string.
+  ///   If `body` has a return value,
+  ///   that value is also used as the return value for this method.
+  /// - Returns: The return value, if any, of the `body` closure parameter.
+  ///
+  /// The pointer passed as an argument to `body` is valid
+  /// only during the execution of this method.
+  /// Don't try to store the pointer for later use.
+  public func withPlatformString<Result>(
+    _ body: (UnsafePointer<PlatformChar>) throws -> Result
+  ) rethrows -> Result {
+    try storage.withPlatformString(body)
+  }
+}
+
+extension FilePath.Component {
+  /// Creates a file path component by copying bytes from a null-terminated platform string.
+  ///
+  /// - Parameter string: A pointer to a null-terminated platform string.
+  public init(platformString: UnsafePointer<PlatformChar>) {
+    self.init(SystemString(platformString: platformString))
+  }
+
+  /// Calls the given closure with a pointer to the contents of the file path component,
+  /// represented as a null-terminated platform string.
+  ///
+  /// If this is not the last component of a path, an allocation will occur in order to
+  /// add the null terminator
+  ///
+  /// - Parameter body: A closure with a pointer parameter
+  ///   that points to a null-terminated platform string.
+  ///   If `body` has a return value,
+  ///   that value is also used as the return value for this method.
+  /// - Returns: The return value, if any, of the `body` closure parameter.
+  ///
+  /// The pointer passed as an argument to `body` is valid
+  /// only during the execution of this method.
+  /// Don't try to store the pointer for later use.
+  public func withPlatformString<Result>(
+    _ body: (UnsafePointer<PlatformChar>) throws -> Result
+  ) rethrows -> Result {
+    try slice.withPlatformString(body)
+  }
+}
+
+
+
+// @available(macOS 10.16, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
+extension FilePath: ExpressibleByStringLiteral {
+  /// Creates a file path from a string literal.
+  ///
+  /// - Parameter stringLiteral: A string literal
+  ///   whose Unicode encoded contents to use as the contents of the path.
+  public init(stringLiteral: String) {
+    self.init(stringLiteral)
+  }
+
+  /// Creates a file path from a string.
+  ///
+  /// - Parameter string: A string
+  ///   whose Unicode encoded contents to use as the contents of the path.
+  public init(_ string: String) {
+    self.init(SystemString(string))
+  }
+}
+
+// @available(macOS 10.16, iOS 14.0, watchOS 7.0, tvOS 14.0, *)
+extension String {
+  /// Creates a string by interpreting the file path's content as UTF-8 on Unix
+  /// and UTF-16 on Windows.
+  ///
+  /// - Parameter path: The file path to be interpreted as `PlatformUnicodeEncoding`.
+  ///
+  /// If the content of the file path isn't a well-formed Unicode string,
+  /// this initializer replaces invalid bytes them with U+FFFD.
```

Comment:
We probably shouldn't leave "invalid bytes them" in the documentation comments. Should this sentence refer to "invalid bytes" or to "them"?

This phrase occurs in a few other places here as well.

Reply:
Good catch. Dropping "them".